http-poja/src/main/java/io/micronaut/http/poja/RawHttpBasedServletHttpResponse.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.poja;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.convert.value.MutableConvertibleValues;
+import io.micronaut.core.convert.value.MutableConvertibleValuesMap;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MutableHttpHeaders;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.cookie.Cookie;
+import io.micronaut.http.simple.SimpleHttpHeaders;
+import io.micronaut.servlet.http.ServletHttpResponse;
+import rawhttp.core.HttpVersion;
+import rawhttp.core.RawHttpHeaders;
+import rawhttp.core.RawHttpResponse;
+import rawhttp.core.StatusLine;
+import rawhttp.core.body.EagerBodyReader;
+
+import java.io.BufferedWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.Optional;
+
+/**
+ * @author Sahoo.
+ */
+class RawHttpBasedServletHttpResponse implements ServletHttpResponse<RawHttpResponse<Void>, String> {
+
+    private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+    private int code = HttpStatus.OK.getCode();
+
+    private String reason = HttpStatus.OK.getReason();
+
+    private String body;
+    private final SimpleHttpHeaders headers;
+
+    private final MutableConvertibleValues<Object> attributes = new MutableConvertibleValuesMap<>();
+
+    public RawHttpBasedServletHttpResponse(ConversionService conversionService) {
+        this.headers = new SimpleHttpHeaders(conversionService);
```

Comment:
`SimpleHttpHeaders` may not be good enough as it doesn't include header validation.

Reply:
The headers are parsed by RawHTTP. I suppose if we replace it, the implementation might be better.